### PR TITLE
fix BUG: ValueError when performing rolling covariance on multi indexed DataFrame

### PR DESCRIPTION
**Pull request by ghost**
- [x] tests added / passed

Proposed fix for #16789. Using class properties (with implemented setter/getter) rather than field variables.

### CODE REVIEW

File: pandas/tests/indexes/test_multi.py [hunk 1]
```diff
@@ -61,6 +61,16 @@ def f():
 
         tm.assert_raises_regex(ValueError, 'The truth value of a', f)
 
+    def test_multi_index_names(self):
+        cols = pd.MultiIndex.from_product([['A', 'B'], ['C', 'D', 'E']],
```
Comment (jreback) [inline pandas/tests/indexes/test_multi.py]: can you add a comment with the issue number

File: pandas/tests/indexes/test_multi.py [hunk 2]
```diff
@@ -61,6 +61,16 @@ def f():
...
+                                          names=['1', '2'])
+        df = pd.DataFrame(np.ones((10, 6)), columns=cols)
+        rolling_result = df.rolling(3).cov()
```
Comment (jreback) [inline pandas/tests/indexes/test_multi.py]: just call this result

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. pls add a whatsnew note in bug fixes / reshaping for 0.20.3. ping on green.

File: pandas/tests/indexes/test_multi.py (hunk 2 above)
Comment (Tuan333) [inline pandas/tests/indexes/test_multi.py]: thanks. will do.

File: pandas/tests/indexes/test_multi.py (hunk 1 above)
Comment (Tuan333) [inline pandas/tests/indexes/test_multi.py]: will do.

File: doc/source/whatsnew/v0.20.3.txt [hunk 3]
```diff
@@ -39,7 +39,7 @@ Bug Fixes
 - Fixed issue with dataframe scatter plot for categorical data that reports incorrect column key not found when categorical data is used for plotting (:issue:`16199`)
 - Fixed a pytest marker failing downstream packages' tests suites (:issue:`16680`)
 - Fixed compat with loading a ``DataFrame`` with a ``PeriodIndex``, from a ``format='fixed'`` HDFStore, in Python 3, that was written in Python 2 (:issue:`16781`)
-
+- Fixed bug in ``pandas.core.window_flex_binary_moment()`` when a multi-index DataFrame is passed in as ``arg1`` (:issue:`16789`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.20.3.txt]: `window_flex_binary_moment` is an internal method, so the user reading these notes doesn't need to care where exactly the bug is. Better to just say something like

    Fixed bug where computing the rolling covariance of a MultiIndexed ``DataFrame`` improperly raised a ``ValueError`` (:issue:`16789`)

Comment (Tuan333) [inline doc/source/whatsnew/v0.20.3.txt]: Thanks. I'll quickly fix it now.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**